WordPressKit/BlazeServiceRemote.swift

```diff
@@ -34,6 +32,33 @@ open class BlazeServiceRemote: ServiceRemoteWordPressComREST {
 
             callback(.failure(error))
         })
+    }
+
+    // MARK: - Campaigns
 
+    /// Searches the campaigns for the site with the given ID. The campaigns are returned ordered by the post date.
+    ///
+    /// - parameters:
+    ///   - siteId: The site ID.
+    ///   - page: The response page. By default, returns the first page.
+    open func searchCampaigns(forSiteId siteId: Int, page: Int = 1, callback: @escaping (Result<BlazeCampaignsSearchResponse, Error>) -> Void) {
```

Comment:
There is a separate [ticket](https://github.com/wordpress-mobile/WordPress-iOS/issues/20794) for adding search API, so I intentionally kept it out of this PR. I don't know yet what the parameters are, but I decided to name it "search" because that's what it will eventually represent.

There is another endpoint for fetching campaigns: `{{base-url}}/wpcom/v2/sites/:siteid/wordads/dsp/api/v1/campaigns/site/:siteid/summary`. I think  @momo-ozawa mentioned that we were going to use the search endpoint for both the dashboard and the campaign list. I'm now a bit unsure about it because:

- It may be worth adding campaigns to [/dashboard/cards-data?cards=todays_stats,posts,pages,activity](https://public-api.wordpress.com/wpcom/v2/sites/219445161/dashboard/cards-data?cards=todays_stats,posts,pages,activity) instead to reduce the number of API calls needed to render dashboard
- Unlike /search, /summary has the following field: `"canCreateCampaigns": true`, which I'm not sure yet what it is for

Reply:
I will rename/update it when the ongoing discussions are resolved. For now, this PR added the data models, which is what we primarily need to the UI.